src/utils/formatBalance.ts

```diff
@@ -62,13 +63,53 @@ export const formatFixedNumber = (number: ethers.FixedNumber, displayDecimals =
   return formatBigNumber(ethers.BigNumber.from(leftSide), displayDecimals, decimals)
 }
 
-export const formatLocalisedCompactNumber = (number: number): string => {
+export const formatLocalisedCompactNumber = (
+  number: number,
+  t: (key: string, data?: ContextData) => string,
+): string => {
   const codeFromStorage = getLanguageCodeFromLS()
+  const hasNotFormatSupport = typeof Intl === 'undefined' || !('NumberFormat' in Intl) || !supportsES2020()
+
+  if (hasNotFormatSupport) {
+    return nFormatter(number, t)
+  }
+
   return new Intl.NumberFormat(codeFromStorage, {
     notation: 'compact',
     compactDisplay: 'long',
     maximumSignificantDigits: 2,
   }).format(number)
 }
 
-export default formatLocalisedCompactNumber
+const nFormatter = (num, t) => {
+  if (num >= 1000000000) {
+    return `${(num / 1000000000).toFixed(1).replace(/\.0$/, '')} ${t('Billion').toLocaleLowerCase()}`
+  }
+  if (num >= 1000000) {
+    return `${(num / 1000000).toFixed(1).replace(/\.0$/, '')} ${t('Million').toLocaleLowerCase()}`
+  }
+  if (num >= 1000) {
+    return `${(num / 1000).toFixed(1).replace(/\.0$/, '')} ${t('Thousand').toLocaleLowerCase()}`
+  }
+  return num
+}
+
+const supportsES2020 = () => {
+  try {
+    const s = new Intl.NumberFormat('en', {
```

Comment:
Please use a more descriptive variable name.

Reply:
Removed, not needed in new approach